src/main/java/backend/Logic.java

```diff
@@ -157,37 +160,84 @@ public void refresh() {
         for (Map.Entry<Integer, IssueMetadata> entry : metadata.entrySet()) {
             IssueMetadata currentMetadata = entry.getValue();
             Date lastNonSelfUpdate = new Date(0);
+            Date lastSelfUpdate = new Date(0);
+            boolean isUpdatedByOthers = false;
+            boolean isUpdatedBySelf = false;
             for (TurboIssueEvent event : currentMetadata.getEvents()) {
-                if (!event.getActor().getLogin().equalsIgnoreCase(currentUser)
-                        && event.getDate().after(lastNonSelfUpdate)) {
+                if (isNewEventByOther(currentUser, event, lastNonSelfUpdate)) {
                     lastNonSelfUpdate = event.getDate();
+                    isUpdatedByOthers = true;
+                }
+                if (isNewEventBySelf(currentUser, event, lastSelfUpdate)){
+                    lastSelfUpdate = event.getDate();
+                    isUpdatedBySelf = true;
                 }
             }
             for (Comment comment : currentMetadata.getComments()) {
-                if (!comment.getUser().getLogin().equalsIgnoreCase(currentUser)
-                        && comment.getCreatedAt().after(lastNonSelfUpdate)) {
+                if (isNewCommentByOther(currentUser, comment, lastNonSelfUpdate)) {
                     lastNonSelfUpdate = comment.getCreatedAt();
+                    isUpdatedByOthers = true;
+                }
+                if (isNewCommentBySelf(currentUser, comment, lastSelfUpdate)){
+                    lastSelfUpdate = comment.getCreatedAt();
+                    isUpdatedBySelf = true;
                 }
             }
 
             entry.setValue(new IssueMetadata(currentMetadata,
                     LocalDateTime.ofInstant(lastNonSelfUpdate.toInstant(), ZoneId.systemDefault()),
-                    calculateNonSelfCommentCount(currentMetadata.getComments(), currentUser)
+                    LocalDateTime.ofInstant(lastSelfUpdate.toInstant(), ZoneId.systemDefault()),
+                    calculateCommentCount(currentMetadata.getComments(), currentUser, UpdatedKind.OTHER_UPDATED),
+                    calculateCommentCount(currentMetadata.getComments(), currentUser, UpdatedKind.SELF_UPDATED),
```

Comment:
Same issue as before. The values of these two arguments can be derived from the others, so you can compute them in the `TurboIssue` constructor. We want to keep `Logic` as high-level as possible.